schematic/schemas/json_schema_validation_rule_functions.py

```diff
@@ -0,0 +1,394 @@
+
+    Arguments:
+        inputted_rule: The full regex rule
+
+    Returns:
+        If the module parameter is search or match, and the pattern parameter exists
+          the pattern is returned
+        Otherwise None
+    """
+    module: Optional[str] = None
+    pattern: Optional[str] = None
+    parameters = _get_parameters_from_inputted_rule(inputted_rule)
+    if parameters:
+        if "module" in parameters:
+            module = parameters["module"]
+        if "pattern" in parameters:
+            pattern = parameters["pattern"]
+    if module is None or pattern is None:
+        return None
+    # Do not translate other modules
+    if module not in [item.value for item in RegexModule]:
+        return None
+    # Match is just search but only at the beginning of the string
+    if module == RegexModule.MATCH.value and not pattern.startswith("^"):
+        return f"^{pattern}"
+    return pattern
+
+
+def get_validation_rule_names_from_inputted_rules(
+    inputted_rules: list[str],
+) -> list[ValidationRuleName]:
+    """Gets a list of ValidationRuleNames from a list of inputted validation rules
+
+    Args:
+        inputted_rules: A list of inputted validation rules form a data model
+
+    Returns:
+        A list of ValidationRuleNames
+    """
+    validation_rules = [
+        _VALIDATION_RULES.get(rule_name_string)
+        for rule_name_string in get_names_from_inputted_rules(inputted_rules)
+    ]
+    return [rule.name for rule in validation_rules if rule is not None]
+
+
+def get_names_from_inputted_rules(inputted_rules: list[str]) -> list[str]:
+    """Gets the names from a list of inputted rules
+
+    Arguments:
+        inputted_rules: A list of inputted validation rules form a data model
+
+    Returns:
+        The names of the inputted rules
+    """
+    return [_get_name_from_inputted_rule(rule) for rule in inputted_rules]
+
+
+def _get_parameters_from_inputted_rule(inputted_rule: str) -> Optional[dict[str, str]]:
+    """Creates a dictionary of parameters and values from and input rule string
+
+    Args:
+        input_rule: An inputted validation rule form a data model
+
+    Returns:
+        If the rule exists, a dictionary where
+          the keys are the rule parameters
+          the values are the input rule parameter values
+        Else None
+    """
+    rule_name = _get_name_from_inputted_rule(inputted_rule)
+    rule_values = inputted_rule.split(" ")[1:]
+    rule = _VALIDATION_RULES.get(rule_name)
+    if rule and rule.parameters:
+        return dict(zip(rule.parameters, rule_values))
+    return None
+
+
+def _get_name_from_inputted_rule(inputted_rule: str) -> str:
+    """Gets the name from an inputted rule
+
+    Arguments:
+        inputted_rule: An inputted validation rule form a data model
+
+    Returns:
+        The name of the inputted rule
+    """
+    return inputted_rule.split(" ")[0]
+
+
+def _get_rules_by_name(names: list[str]) -> list[ValidationRule]:
```

Comment:
nit: should this function be called `_get_rules_by_names` since the input and output are both plural?